### PR TITLE
lvfntman: add error traces for some freetype calls

### PR DESCRIPTION
**Pull request by benoit-pierre**
This change is [Reviewable](https://reviewable.io/reviews/koreader/crengine/573)

### CODE REVIEW

**Comment by poire-z** [comment]
Why did you need that, and what do you/we can expect to see? And how often may we see them?
For CRE explicite high-level-business warn/error that may matter to the user and us, I added some printf with some "CRE" prefix so they stand out. So may be a hardcoded CRE_FT prefix would be nice.
crengine has its own CRLOG stuff (that I rarely need to enable) - dunno if it uses stdout or stderr; it feels odd to add a third system of logging.

File: crengine/src/lvfntman.cpp [hunk 1]
```diff
@@ -2015,9 +2025,11 @@ class LVFreeTypeFace : public LVFont
         _fontFamily = fontFamily;
         if (_face)
             FT_Done_Face(_face);
-        int error = FT_New_Memory_Face( _library, buf->get(), buf->length(), index, &_face ); /* create face object */
-        if (error)
+        FT_Error error = FT_New_Memory_Face( _library, buf->get(), buf->length(), index, &_face ); /* create face object */
+        if (FT_Err_Ok != error) {
```
Comment (poire-z) [inline crengine/src/lvfntman.cpp:2029]: These kind of reverse comparisons (I know they are supposed to help with typos or forgetting a `=` in `==`, but are harder to read, I needed to pause 10s before swaping they in my head :)) are rare in crengine.

Comment (benoit-pierre) [reply crengine/src/lvfntman.cpp:2029]: Another habit ¯\\_(ツ)_/¯. There were already others in that same file.

**Comment by benoit-pierre** [comment]
> Why did you need that, and what do you/we can expect to see? And how often may we see them?

When some obfuscated fonts can be loaded for example :P. So not often. But since an ebook might still appear to render fine, I think it's a good idea to have something in the log warning us about it.

> For CRE explicite high-level-business warn/error that may matter to the user and us, I added some printf with some "CRE" prefix so they stand out. So may be a hardcoded CRE_FT prefix would be nice.

I'll add a prefix.

> crengine has its own CRLOG stuff (that I rarely need to enable) - dunno if it uses stdout or stderr; it feels odd to add a third system of logging.

Yeah, but that's a lot of baggage, and it's not enabled by default (even in debug mode). We already have some `fprintf(…)` calls sprinkled here and there.

**Comment by benoit-pierre** [comment]
The font related errors I would get rid of are about already registered fonts: my log is littered with those (including when running the test suite).

**Comment by poire-z** [comment]
> my log is littered with those

So, why does this happen ? Why do you keep these fonts that crengine can't use ? :)
I think it's good to let the user know some won't be used (happened to me I think when playing with multi-weighted font, it's good to see which aren't used).

**Pull request merged**